src/integration/transaction/application/services/__tests__/transaction.service.spec.ts

```diff
@@ -0,0 +1,120 @@
+  });
+
+  it('should return a tx for open, if a tx got created', () => {
+    setup(TestSetup.CREATE_MASTERNODE);
+    expect(service.getOpen()).toMatchObject([
+      {
+        issuerSignature: 'create-masternode-signed-message',
+        rawTx: {
+          hex: 'create-masternode',
+          scriptHex: '0',
+          prevouts: [],
+        },
+        payload: {
+          ownerWallet: 'cold-wallet-a',
+          accountIndex: 1,
+        },
+      },
+    ]);
+  });
+
+  it('should return an empty array for open, if a tx got created and verified', () => {
+    setup(TestSetup.CREATE_MASTERNODE);
+    const txId = service.getOpen()[0].id;
+    service.verify(txId, 'verifier-signed-message');
+    expect(service.getOpen()).toStrictEqual([]);
+  });
+
+  it('should return an empty array for open, if a tx got created, verified and signed', () => {
+    setup(TestSetup.CREATE_MASTERNODE);
+    const txId = service.getOpen()[0].id;
+    service.verify(txId, 'verifier-signed-message');
+    service.signed(txId, 'signed-raw-tx-hex');
+    expect(service.getOpen()).toStrictEqual([]);
+  });
+
+  it('should return an empty array for verified, if no tx is created', () => {
+    setup(TestSetup.EMPTY);
+    expect(service.getVerified()).toStrictEqual([]);
+  });
+
+  it('should return an empty array for verified, if a tx got created', () => {
+    setup(TestSetup.CREATE_MASTERNODE);
+    expect(service.getVerified()).toStrictEqual([]);
+  });
+
+  it('should return a tx for verified, if a tx got created and verified', () => {
+    setup(TestSetup.CREATE_MASTERNODE);
+    const txId = service.getOpen()[0].id;
+    service.verify(txId, 'verifier-signed-message');
+    expect(service.getVerified()).toMatchObject([
+      {
+        issuerSignature: 'create-masternode-signed-message',
+        verifierSignature: 'verifier-signed-message',
+        rawTx: {
+          hex: 'create-masternode',
+          scriptHex: '0',
+          prevouts: [],
+        },
+        payload: {
+          ownerWallet: 'cold-wallet-a',
+          accountIndex: 1,
+        },
+      },
+    ]);
+  });
+
+  it('should return an empty array for verified, if a tx got created, verified and signed', () => {
+    setup(TestSetup.CREATE_MASTERNODE);
+    const txId = service.getOpen()[0].id;
+    service.verify(txId, 'verifier-signed-message');
+    service.signed(txId, 'signed-raw-tx-hex');
+    expect(service.getVerified()).toStrictEqual([]);
+  });
+
+  it('should throw an exception (NotFoundException), if a tx can not be found on verify', () => {
+    setup(TestSetup.CREATE_MASTERNODE);
+    expect(() => service.verify('some-tx-which-is-not-found', 'some-signature')).toThrow(NotFoundException);
+  });
+
+  it('should throw an exception (NotFoundException), if a tx can not be found on signed', () => {
+    setup(TestSetup.CREATE_MASTERNODE);
+    expect(() => service.signed('some-tx-which-is-not-found', 'some-signed-raw-hex')).toThrow(NotFoundException);
+  });
+});
```

Comment:
What about a test case for the sign method? Something like
```ts
setTimeout(() => {
  service.verify(txId, 'verifier-signed-message');
  service.signed(txId, 'signed-raw-tx-hex');
}, 100);

const hex = await service.sign(...);
```

Reply:
ähm yeah would be cool, but where do I receive `txId` from? this will be generated inside `service.sign`.

Reply:
Hm good point. We would need to implement dummy verifiers and signers 🤔